### PR TITLE
Add call to irm at market creation

### PR DESCRIPTION
**Pull request by Jean-Grimal**
Fixes https://cantina.xyz/ai/d86b7f95-e574-4092-8ea2-78dcac2f54f1/findings/3a235a1b-ddb4-4f7f-b254-0c940a832814

This issue doesn't really have a meaningful impact, so we could just acknowledge it as well.

### CODE REVIEW

**Comment by QGarchery** [comment]
Updated so that it does not contain the commits on main. The CI is expected to fail on Certora because now there are other ways the `createMarket` function can revert

**Review by Rubilmax — APPROVED** [review]

File: src/Morpho.sol [hunk 1]
```diff
@@ -158,6 +158,9 @@ contract Morpho is IMorphoStaticTyping {
         idToMarketParams[id] = marketParams;
 
         emit EventsLib.CreateMarket(id, marketParams);
+
+        // Call IRM in case it is stateful.
```
Comment (Rubilmax) [inline src/Morpho.sol:162]:
```suggestion
        // Call to initialize IRM in case it is stateful.
```

**Review by Rubilmax — APPROVED** [review]

File: src/Morpho.sol [hunk 2]
```diff
@@ -158,6 +158,9 @@ contract Morpho is IMorphoStaticTyping {
         idToMarketParams[id] = marketParams;
 
         emit EventsLib.CreateMarket(id, marketParams);
+
+        // Call to initialize IRM in case it is stateful.
```
Comment (MerlinEgalite) [inline src/Morpho.sol:162]:
```suggestion
        // Call to initialize the IRM in case it is stateful.
```

File: src/Morpho.sol [hunk 3]
```diff
@@ -158,6 +158,9 @@ contract Morpho is IMorphoStaticTyping {
         idToMarketParams[id] = marketParams;
 
         emit EventsLib.CreateMarket(id, marketParams);
+
+        // Call to initialize the IRM in case it is stateful.
+        IIrm(marketParams.irm).borrowRate(marketParams, market[id]);
```
Comment (MathisGD) [inline src/Morpho.sol:163]: maybe it's better now to call _accrueInterest?

Comment (Rubilmax) [reply src/Morpho.sol:163]: It will be skipped because `elapsed == 0`

Comment (MathisGD) [reply src/Morpho.sol:163]: No, elapsed is very big instead (we would not have touched lastUpdated at all for the moment). In any case I agree that it's weird, let me think about a nice fix.

Comment (MathisGD) [reply src/Morpho.sol:163]: pushed in #642

**Comment by MerlinEgalite** [comment]
Why do we bundle different fixes in this PR @MathisGD ?

**Comment by MathisGD** [comment]
Oops it was not meant, feel free to force push

**Comment by Rubilmax** [comment]
> Oops it was not meant, feel free to force push

done

**Comment by MathisGD** [comment]
but my commits are still here

**Review by Rubilmax — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Pull request merged**